File: BE/src/main/java/com/ssafy/e106/domain/subscription/dto/response/SubscriptionCoveredServiceResponse.java
package com.ssafy.e106.domain.subscription.dto.response;

import com.ssafy.e106.domain.subscription.entity.Service;

public record SubscriptionCoveredServiceResponse(
    Long serviceId,
    String serviceCode,
    String serviceName,
    String logoUrl) {

  public static SubscriptionCoveredServiceResponse from(Service service) {
    return new SubscriptionCoveredServiceResponse(
        service.getServiceId(),
        service.getCode(),
        service.getName(),
        service.getLogoUrl());
  }
}

File: FE/app/src/main/java/com/ssafy/e106/data/dto/subscription/SubscriptionCoveredServiceResponse.kt
package com.ssafy.e106.data.dto.subscription

import kotlinx.serialization.Serializable

@Serializable
data class SubscriptionCoveredServiceResponse(
    val serviceId: Long,
    val serviceCode: String? = null,
    val serviceName: String,
    val logoUrl: String? = null,
)
